src/darsia/analysis/binarymassanalysis.py

```diff
@@ -0,0 +1,232 @@
+    ) -> float:
+        """
+        Given a segmentation, external pressure and the number
+        associated to free CO2 (default is 2), it returns the
+        mass of free CO2 in grams.
+
+        Arguments:
+            segmentation (np.ndarray): segmented image
+            external_pressure (float): external pressure, for example atmospheric pressure.
+            co2_component (int): value of the component in segmentation that
+            corresponds to co2.
+
+        Returns:
+            (float): Mass of co2.
+        """
+        volume = self.volume_map(segmentation, co2_component)
+        density_map = self.external_pressure_to_density_co2(external_pressure)
+        return np.sum(volume * density_map)
+
+    def _compute_depth_map(
+        self, depth_measurements: tuple[np.ndarray, ...]
+    ) -> np.ndarray:
+        """
+        Compute depth map, based on the reported measurements.
+
+        Arguments:
+             depth_measurements (Optional[tuple[np.ndarray, ...]]): tuple of depth
+                    measurements. Should be provided as horizontal coordinates,
+                    vertical coordinates, and depth cooridinates corresponding
+                    to each horizontal and vertical entry.
+        Returns:
+            (np.ndarray): depth map
+        """
+
+        # Determine number of voxels in each dimension - assume 2d image
+        Ny, Nx = self.base_image.img.shape[:2]
+        x = np.arange(Nx)
+        y = np.arange(Ny)
+        X_pixel, Y_pixel = np.meshgrid(x, y)
+        pixel_vector = np.transpose(np.vstack((np.ravel(Y_pixel), np.ravel(X_pixel))))
+        coords_vector = self.base_image.coordinatesystem.pixelToCoordinate(pixel_vector)
+        # Fetch physical dimensions
+        # Depth of the rig, measured in discrete points, taking into account expansion.
+        # Values taken from the benchmark description.
+        # Coordinates at which depth measurements have been taken.
+        # Note that the y-coordinate differs depending on the x-coordinate,
+        # which dissallows use of np.meshgrid. Instead, the meshgrid is
+        # constructed by hand.
+        depth_interpolator = RBFInterpolator(
+            np.transpose(
+                np.vstack(
+                    (
+                        depth_measurements[0],
+                        depth_measurements[1],
+                    )
+                )
+            ),
+            depth_measurements[2],
+        )
+        # Evaluate depth function to determine depth map
+        depth_vector = depth_interpolator(coords_vector)
+        return depth_vector.reshape((Ny, Nx))
+
+    def write_depth_map(self, path: Union[str, Path]) -> None:
+        """
+        Writes the depth map to file (It is quite time consuming to compute).
+
+        Arguments:
+            path (Union[str, Path]): location where depthmap will be saved.
+
+        TODO: Might be an unnecessary function.
```

Comment:
As long there is no corresponding read routine, it indeed looks redundant. I think in order for this to make sense, there should be an option for reading a depth map from file in the constructor. Having a look at the input arguments of the constructor, this will become messy to have three different arguments which have different type and it is implicitly decided which of the options to prioritize in the unlikely event that several options are chosen. Maybe one should instead move them to kwargs and instead add a string as true input which reads what type of input is provided, and then the correct keywords are checked to fill in the remaining info?

Reply:
In such a case I would find it natural to envoke the writing routine in the constructor if such a file for given path did not exist and could not be read. If it does, read from file using np.load. Actually in this case I would also include this one line (the method is nothing more than that just with another name) directly in the constructor.